service/src/test/java/com/codeforcommunity/processor/TeamsProcessorImplTest.java

```diff
@@ -0,0 +1,420 @@
+  Emailer emailer;
+
+  /** Method to setup mockDb and processor. */
+  @BeforeEach
+  void setup() {
+    mockDb = new JooqMock();
+    emailer = mock(Emailer.class);
+    processor = new TeamsProcessorImpl(mockDb.getContext(), emailer);
+  }
+
+  void userCreatingTeam() {
+    UsersRecord myUser = mockDb.getContext().newRecord(Tables.USERS);
+    myUser.setUsername("kiminusername");
+    myUser.setEmail("kimin@example.com");
+    myUser.setPassHash(Passwords.createHash("password"));
+    myUser.setId(1);
+    myUser.setPrivilegeLevel(PrivilegeLevel.STANDARD);
+    mockDb.addReturn("SELECT", myUser);
+    jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+  }
+
+  void team1() {
+    TeamRecord myTeam = mockDb.getContext().newRecord(Tables.TEAM);
+    myTeam.setId(5);
+    myTeam.setName("teamKimin");
+    myTeam.setBio("bioTeam");
+    myTeam.setGoal(2);
+    Timestamp timestamp = new Timestamp(System.currentTimeMillis());
+    myTeam.setGoalCompletionDate(timestamp);
+    mockDb.addReturn("SELECT", myTeam);
+  }
+
+  // successfully create a team
+  @Test
+  public void testCreateTeam1() {
+    mockDb.addEmptyReturn("SELECT");
+    mockDb.addEmptyReturn("INSERT");
+
+    Timestamp timestamp = Timestamp.valueOf("2020-05-30 02:00:55.939");
+    TeamInvitationRequest ex1 = new TeamInvitationRequest("ex1", "ex1@example.com");
+    TeamInvitationRequest ex2 = new TeamInvitationRequest("ex2", "ex2@example.com");
+    TeamInvitationRequest ex3 = new TeamInvitationRequest("ex3", "ex3@example.com");
+    List<TeamInvitationRequest> emailList = Arrays.asList(ex1, ex2, ex3);
+    CreateTeamRequest teamRequest =
+        new CreateTeamRequest("teamName", "teamBio", 2, timestamp, emailList);
+
+    jwtData = new JWTData(1, PrivilegeLevel.STANDARD);
+    processor.createTeam(jwtData, teamRequest);
+    assertEquals(2, mockDb.timesCalled("SELECT"));
+    assertEquals(2, mockDb.timesCalled("INSERT"));
+
+    ArgumentCaptor<String> stringArgs = ArgumentCaptor.forClass(String.class);
+    ArgumentCaptor<Users> usersArgs = ArgumentCaptor.forClass(Users.class);
+    ArgumentCaptor<Team> teamArgs = ArgumentCaptor.forClass(Team.class);
+    verify(emailer, times(3))
+        .sendInviteEmail(
+            stringArgs.capture(), stringArgs.capture(), usersArgs.capture(), teamArgs.capture());
+
+    List<String> capturedStrings = stringArgs.getAllValues();
+    assertEquals("ex1@example.com", capturedStrings.get(0));
+    assertEquals("Team Member", capturedStrings.get(1));
+    assertEquals("ex2@example.com", capturedStrings.get(2));
+    assertEquals("Team Member", capturedStrings.get(3));
+    assertEquals("ex3@example.com", capturedStrings.get(4));
+    assertEquals("Team Member", capturedStrings.get(5));
+    List<Users> capturedUsers = usersArgs.getAllValues();
+    // code should be fixed
+    // assertEquals("test", capturedUsers.get(0));
+    //        assertEquals("null", capturedUsers.get(1));
+    //        assertEquals("null", capturedUsers.get(2));
```

Comment:
What's up here again?

Reply:
I found a bug where the inviter data/information is not properly stored when sendInviteEmail is called during the createTeam() function. I have made a bug report in the past at 
https://docs.google.com/document/d/1o-Puae_LjAZjLFpLvQ0EUZ4fXTp4qYhQf9YR0JmmYl0/edit?usp=sharing